BayeuxClient.agent.lib.nut

```diff
@@ -0,0 +1,936 @@
+                subscribes.append(msg);
+            } else if (msg["channel"].find(BAYEUX_CLIENT_META_CHANNEL_UNSUBSCRIBE) == 0) {
+                unsubscribes.append(msg);
+            } else {
+                others.append(msg);
+            }
+        }
+        messages.clear();
+        messages.extend(subscribes);
+        messages.extend(others);
+        messages.extend(unsubscribes);        
+    }
+
+    function _onUserMessage(message) {
+        local channelPart = "";
+        // For example: message["channel"] = "/a/b/c". Then channelParts = ["a", "b", "c"]
+        local channelParts = split(message["channel"], "/");
+        // Example: partsNum = 3
+        local partsNum = channelParts.len();
+        local channelLen = 0;
+        local builtChannel = null;
+        // Example: 3 iterations
+        foreach (part in channelParts) {
+            // Example: 
+            // 1-st iter: channelPart = "/a", channelLen = 1
+            // 2-nd iter: channelPart = "/a/b", channelLen = 2
+            // 3-rd iter: channelPart = "/a/b/c", channelLen = 3
+            channelPart += "/" + part;
+            channelLen++;
+            // Example: this branch on the 3-rd iteration
+            if (channelLen == partsNum) {
+                // Example: builtChannel = "/a/b/c"
+                builtChannel = channelPart;
+            // Example: this branch on the 2-nd iteration
+            } else if (channelLen == partsNum - 1) {
+                // Example: builtChannel = "/a/b/*"
+                builtChannel = channelPart + "/*";
+
+                if (builtChannel in _userHandlers) {
+                    // TODO: User's handler can take a lot of time, so we can lose a connection due to timeout. 
+                    // Should we do imp.wakeup(0, ...) here to allow the lib firstly send "connect" message and only then call user's handler?
+                    _userHandlers[builtChannel](message["channel"], message["data"]);
+                }
+
+                // Example: builtChannel = "/a/b/**"
+                builtChannel += "*";
+            // Example: this branch on the 1-st iteration
+            } else {
+                // Example: builtChannel = "/a/**"
+                builtChannel = channelPart + "/**";
+            }          
+
+            if (builtChannel in _userHandlers) {
+                // TODO: Same as above (imp.wakeup)
+                _userHandlers[builtChannel](message["channel"], message["data"]);
+            }
+        }
+    }
+
+    function _handshake(callback) {
+        local msgId = _nextMessageId();
+        local message = {
+            "id" : msgId,
+            "version" : BAYEUX_CLIENT_PROTOCOL_VERSION,
+            "minimumVersion" : BAYEUX_CLIENT_PROTOCOL_VERSION,
+            "channel" : BAYEUX_CLIENT_META_CHANNEL_HANDSHAKE,
+            "supportedConnectionTypes" : [BAYEUX_CLIENT_LONGPOLL_TRANSPORT_NAME],
+            "advice" : {
+                "timeout" : _advice.timeout,
```

Comment:
why not convert it to seconds here?

Reply:
Bayeux protocol uses milliseconds, so we have to deal we them in some places (

Reply:
got it